src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -3501,6 +3539,18 @@ static QVariant fcnArrayGet( const QVariantList &values, const QgsExpressionCont
   return list.at( pos );
 }
 
+static QVariant fcnArrayFirst( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent )
+{
+  const QVariantList list = QgsExpressionUtils::getListValue( values.at( 0 ), parent );
+  return list.value( 0 );
```

Comment:
Oh yeah. I don't like value because it's behavior is non obvious to me. But whatever - the test is there, that's the main thing.